File: core/src/main/java/org/haas/core/domain/movie/repository/MovieRepository.java
package org.haas.core.domain.movie.repository;

import org.haas.core.domain.movie.Movie;

import java.util.List;

public interface MovieRepository {

    Movie findById(Long id);
    Movie save(Movie movie);

    List<Movie> findAllStatusShowingMovies();

    List<Movie> findAllByMovieStatusOrderByReleasedAtDesc();
}

File: core/src/main/kotlin/kr/spartacodingclub/cinema/core/repository/MovieRepository.kt
package kr.spartacodingclub.cinema.core.repository

import kr.spartacodingclub.cinema.core.domain.Movie
import kr.spartacodingclub.cinema.core.domain.Theater
import java.time.LocalDateTime

interface MovieRepository {
    fun findAllOrderByReleaseDateDesc(): List<Movie>
    fun findById(id: Long): Movie?
    fun findAllWithScreeningsOrderByReleaseDateDesc(): List<MovieWithScreenings>
}

data class MovieWithScreenings(
    val movie: Movie,
    val screenings: List<MovieScreening>
)

data class MovieScreening(
    val theater: Theater,
    val startTime: LocalDateTime
)
